aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/MqttManagerIntegrationTest.java

```diff
@@ -115,13 +101,9 @@ public void afterEach() {
 
     @Test
     public void mqttConnect_USEast1_ATS_Endpoint() throws Exception {
-        DescribeEndpointRequest request = new DescribeEndpointRequest();
-        request.setEndpointType("iot:Data-ATS");
-        DescribeEndpointResult result = iotClient.describeEndpoint(request);
-        endpointPrefix = result.getEndpointAddress().split("\\.")[0];
-        System.out.println("Endpoint prefix: " + endpointPrefix);
-        
-        String endpoint = endpointPrefix + ".iot." + Region.getRegion(Regions.US_EAST_1).getName() + ".amazonaws.com";
+        String prefix = iotClient.getEndpointPrefix("iot:Data-ATS");
+        String region = Regions.US_EAST_1.getName();
+        String endpoint = String.format("%s.iot.%s.amazonaws.com", prefix, region);
```

Comment:
Is this logic that can be encapsulated within `IotClient`?

Reply:
Looking at the [docs](https://docs.aws.amazon.com/cli/latest/reference/iot/describe-endpoint.html), it looks like this is what gets returned from `getEndpointAddress()`. Perhaps these can be replaced with a call to that?

Reply:
The `iot:Data-ATS` one is supposed to have `<prefix>-ats.iot.amazonaws.com`, not `<prefix>.iot.amazonaws.com`, so I'm not sure what this business is about.

Since its passing, I'll leave the logic alone for now, and circle back to this once I'm seeing more green tests.